code_samples/api/migration/src/Command/MigrationCommand.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+namespace App\Command;
+
+use Ibexa\Platform\Migration\MigrationService;
+use Ibexa\Platform\Migration\Repository\Migration;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+
+class MigrationCommand extends Command
+
+{
+    private $migrationService;
+
+    public function __construct(MigrationService $migrationService, ?string $name = null)
+    {
+        $this->migrationService = $migrationService;
+        parent::__construct($name);
+    }
+
+    protected function configure()
+    {
+        $this->setName('doc:migration');
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $string_with_migration_content = '';
+        $this->migrationService->add(new Migration(
+                'new_migration.yaml',
+                $string_with_migration_content)
+        );
+
+        foreach ($this->migrationService->listMigrations() as $migration) {
+            $output->writeln($migration->getName());
+        }
+
+        $migration_name = $this->migrationService->listMigrations()[0]->getName();
+        $my_migration = $this->migrationService->findOneByName($migration_name);
```

Comment:
Note: `$this->migrationService->listMigrations()[0]` returns the same object (but not instance :exclamation:) as `$this->migrationService->findOneByName()`

So basically one or the other should be used. You either know the migration name, or you don't :smile:  

```suggestion
        // Get the first migration by querying all migrations.
        $migration = $this->migrationService->listMigrations()[0];
        
        // Get migration by name.
        $migration_name = $migration->getName();
        $my_migration = $this->migrationService->findOneByName($migration_name);
```